generators/chipyard/src/main/scala/IOBinders.scala

```diff
@@ -182,6 +183,20 @@ object AddIOCells {
     port.suggestName("serial")
     (port, ios)
   }
+
+  def axi4(io: Seq[AXI4Bundle], node: AXI4SlaveNode): Seq[(AXI4Bundle, AXI4EdgeParameters, Seq[IOCell])] = {
+    io.zip(node.in).map{ case (mem_axi4, (_, edge)) => {
+      val (port, ios) = IOCell.generateIOFromSignal(mem_axi4, Some("iocell_mem_axi4"))
+      port.suggestName("mem_axi4")
+      (port, edge, ios)
+    }}
+  }
+
+  def blockDev(bdev: BlockDeviceIO): (BlockDeviceIO, Seq[IOCell]) = {
+    val (port, ios) = IOCell.generateIOFromSignal(bdev, Some("iocell_bdev"))
+    port.suggestName("bdev")
+    (port, ios)
+  }
```

Comment:
This isn't totally related to your changes, but I wonder if this pattern of adding stuff into the AddIOCells object is the best way to do this moving forwards.
These functions should be able to be inlined into the IOBInders themselves, thus localizing all the code for any IO to one place?

Reply:
Yeah I initially did that but then I needed the same sequence of code in both version of the `Sim` and `Model` devices so I hoisted them to make it DRY.

Reply:
Yeah, the reason for this pattern is that you often need to reuse this part of the code for multiple `IOBinder`s.